### DISCUSSION TITLE
Bad encoding in query parameters with new TestClient using httpx.Client

### DISCUSSION THREAD

**Discussion by azogue** [fastapi/fastapi#6190] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import logging

from fastapi import FastAPI

app = FastAPI()


@app.get("/example")
async def _show_encoding_error(look_for: str):
    return {"found": look_for}


if __name__ == '__main__':
    from fastapi.testclient import TestClient

    with TestClient(app) as client:
        params = {"look_for": "plain text"}
        resp = client.get("/example", params=params).json()
        logging.warning(resp)
        assert resp["found"] == "plain text"

        params = {"look_for": "España"}
        resp = client.get("/example", params=params).json()
        logging.warning(resp)
        assert resp["found"] == "España", resp["found"]
```

### Description

After the change to `httpx` for the `TestClient` in **v0.87.0**, the query parameters are not properly encoded? when sending requests with it, and strings are corrupted when received in the endpoints.

The example app works as expected if called from the SwaggerUI or from another python process using a _plain_ `httpx.Client`, so it appears something broke with the new wrapping for `TestClient` 🥲

```python
import httpx

params = {"look_for": "España"}
with httpx.Client(base_url="http://localhost:8000/") as client:
    resp = client.get("/example", params=params).json()
    assert resp["found"] == "España"
```

### Operating System

macOS

### Operating System Details

M1, running arm64 arch

### FastAPI Version

0.87.0

### Python Version

Python 3.10.5

### Additional Context

starlette-0.21
httpx-0.23.0

Discovered when trying to migrate the test suite for a ~big project previously using fastapi-0.85.1 + starlette-0.20.4. 

All minor syntax changes from old `requests` to new `httpx` were under control, but in one unit test, **a string with an accent** was making some search to fail without results (test is sending "Formalización" but endpoint is receiving **"FormalizaciÃ³n"** 😱), and I was getting crazy 😅

**Comment by azogue** [comment]
I just saw **this it is not related to FastAPI but to Starlette** 😅, after https://github.com/encode/starlette/pull/1376, and the example code above fails the same way if using the original object `from starlette.testclient import TestClient`

I'll try to ask there 👍

Not sure if closing this one, as `starlette` version is pinned to [`0.21.0`](https://github.com/encode/starlette/releases/tag/0.21.0) because of the `TestClient` change

**Comment by azogue** [comment]
For those in a hurry, or just to quick-check that the _strange broken tests_ you're having after the `TestClient` change to `httpx` are related to this, 

`pip install https://lexic-public-assets.s3.eu-west-1.amazonaws.com/wheels/universal/starlette-0.21.0-py3-none-any.whl`

will swap the published version with a fixed one containing https://github.com/encode/starlette/pull/1953 👍

**Comment by Kludex** [comment, marked as answer]
This is solved in Starlette 0.22.0.

**Comment by mezhaka** [comment]
@Kludex What is the plan to fix it in FastAPI? Is there a PR that pulls in the Starlette 0.22.0? The Starlette 0.22.0 is already out.

**Comment by azogue** [comment]
> @Kludex What is the plan to fix it in FastAPI? Is there a PR that pulls in the Starlette 0.22.0? The Starlette 0.22.0 is already out.

@mezhaka ,

in https://github.com/tiangolo/fastapi/pull/5659, but I'm _new_ here 😅, so not sure about the workflow to generate patches, sorry

**Comment by mezhaka** [comment]
@azogue Nice, even with a test!

**Comment by tiangolo** [comment]
Thanks @azogue for reporting and upgrading it here! :cake: 

Thanks everyone for the discussion too! :coffee: 

This is available in FastAPI `0.88.0`, just released. :rocket:
